### DISCUSSION TITLE
Why did you wait to upgrade from 0.x to Restreamer v2?

### DISCUSSION THREAD

**Discussion by svenerbeck** [datarhei/restreamer#363] [Q&A]
Hello

A month ago, we released the Restreamer v2. Thank you for the emails, GitHub issues, and discords you sent us. We hope everyone will enjoy the new software in their daily streaming life. We would now like to know a few things.

- **Where are you having problems switching to the new datarhei Restreamer?**
-  **What are the reasons why the upgrade was not acceptable after a test?** 

We are grateful for any hints to facilitate the change to the latest version.

Thank you for being so helpful. 
datarhei.com Team (SE)

**Comment by utilsites** [comment]
Didn't try yet, because will need to change source code to support secure stream with token (jwt), and also giving some time to have a more stable version with bugs fixed.

**Reply by svenerbeck** [reply]
Hello utilities
Thank you for your honest answer. I can hear your response regarding the use in a commercial application. It's normal behavior for an application, and I wouldn't do it any other way. Customization costs time, money, and nerves! I can already tell you that we have implemented Auth0. You will like that when the update comes. Stay tuned!
[electronic handshake]
Sven

**Comment by bluemantwo** [comment]
I finally upgraded to v2, and an thrilled with the new version! Main improvement for me was multiple streams. And the entire interface is much easier to use and more sexy. Very polished.

Only issue I ran into was use of SSL certs. I am using Lets Encrypt on another server in my home (Home Assistant) to generate cert.pem and key.pem. I could not figure out how to get Restreamer to generate its own Lets Encrypt .pem's and have SSL work. So I just added ENV entries to point to my existing .pem files which I store on my server, and it now works. I should probably go back and figure out whwat I was doing wrong and use the built in Lets Encrypt you have added, but for now I am super happy!

@bluemantwo	Write a reply

**Reply by svenerbeck** [reply]
Hello @bluemantwo 

Thank you for your positive words.

Typically, the SSL hardly causes any problems. However, it is essential always to pay close attention to the ports. 

Maybe these docs and YouTube videos will help you. Otherwise, I recommend opening a separate ticket for the SSL case.

[https://docs.datarhei.com/restreamer/knowledge-base/user-guides/how-can-i-use-https-with-lets-encrypt](https://docs.datarhei.com/restreamer/knowledge-base/user-guides/how-can-i-use-https-with-lets-encrypt)

Have fun streaming with the Restreamer.

Cheerio
Sven
